src/lib/generate-users-csv.js

```diff
@@ -0,0 +1,24 @@
+const Json2csvParser = require('json2csv').Parser;
+const fs = require('fs');
+const info = require('./log').info;
+
+
+ 
+module.exports = (data, csvPath) => {
+    const fields = ['username', 'password', 'roles','contact','phone','place'];
+    const json2csvParser = new Json2csvParser({ fields, doubleQuote: "'", flatten:true});
+    data.forEach(function(user){
+        if (typeof user.contact === 'object') {
+            user.contact = user.contact._id;
+        }
+        //type is deprecated in the api and is replaced by roles
+        //passing type to the api will use that for the roles instead of roles
+        //type here is being generated based on the type of doc being created
+        delete user.type;
+    });
+    const csv = json2csvParser.parse(data);
+    fs.writeFile(csvPath, csv, function(err){
+        if (err) { throw err; }
+        info('Users csv has been saved');
+    });
+};
```

Comment:
Indentation should be 2 spaces not 4. You may need to tweak the default settings in your IDE.

Reply:
I updated a couple things to match our style-guide.